File: Data Structures & Algorithms/count-good-nodes-in-binary-tree/submission-2.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
from collections import deque

class Solution:
    def goodNodes(self, root: TreeNode) -> int:
        res = 0
        queue = deque()

        if root :
            queue.append((root, float("-inf")))
        
        while queue:
            node, maxVal = queue.popleft()

            if node.val >= maxVal:
                res+=1
            
            if node.left:
                queue.append((node.left, max(maxVal, node.val)))
            
            if node.right:
                queue.append((node.right, max(maxVal, node.val)))

        return res

File: Data Structures & Algorithms/binary-tree-preorder-traversal/submission-2.kt
/**
 * Definition for a binary tree node.
 * class TreeNode(var `val`: Int) {
 *     var left: TreeNode? = null
 *     var right: TreeNode? = null
 * }
 */

class Solution {
    fun preorderTraversal(root: TreeNode?): List<Int> {
        val res = mutableListOf<Int>()
        fun preorder(node: TreeNode?){
            if(node == null){
                return
            }
            res.add(node.`val`)
            preorder(node.left)
            preorder(node.right)
        }
        preorder(root)
        return res
    }
}
